src/main/java/com/sprint/mission/discodeit/service/basic/BasicChannelService.java

```diff
@@ -118,17 +95,42 @@ public void delete(UUID channelId) {
             throw new NoSuchElementException("Channel with id " + channelId + " not found");
         }
         // 관련된 도메인도 같이 삭제합니다.
-        messageRepository.findAll().stream()
-                .filter(msg -> msg.getChannelId().equals(channelId))
-                    .forEach(msg -> {
-                        messageRepository.deleteById(msg.getId());
-                    });
-        readStatusRepository.findAll().stream()
-                        .filter(status -> status.getChannelId().equals(channelId))
+        messageRepository.findAllByChannelId(channelId)
+                                .forEach(msg -> {
+                                    messageRepository.deleteById(msg.getId());
+                                });
+        readStatusRepository.findAllByChannelId(channelId)
                                 .forEach(status -> {
                                     readStatusRepository.delete(status.getId());
                                 });
 
         channelRepository.deleteById(channelId);
     }
+
+    private ChannelDto toDto(Channel channel) {
+        Instant lastMessageAt = messageRepository.findAllByChannelId(channel.getId())
+                .stream()
+                .sorted(Comparator.comparing(Message::getCreatedAt).reversed())
+                .map(Message::getCreatedAt)
+                .limit(1)
+                .findFirst()
+                .orElse(Instant.MIN);
+
```

Comment:
```suggestion
   Instant lastMessageAt = messageRepository.findAllByChannelId(channel.getId())
    .stream()
    .map(Message::getCreatedAt)
    .max(Comparator.naturalOrder())
    .orElse(Instant.MIN);
                .orElse(Instant.MIN);

```

limit 에 findFirst 까지하면 의미가 중복되는게 아닌가 싶어서 제안드립니다.